src/java/com/palantir/cassandra/db/compaction/CompactionThroughputThrottler.java

```diff
@@ -11,15 +12,39 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
 /**
  * Backpressure mechanism to reduce compaction throughput for large tables as disk fills up
  */
-public class CompactionThroughputThrottler
+public final class CompactionThroughputThrottler
 {
     private static final boolean ENABLE_COMPACTION_THROUGHPUT_THROTTLING = Boolean.getBoolean("palantir_cassandra.enable_compaction_throughput_throttling");
     private static final long THROTTLER_TABLE_SIZE_BYTES = Long.getLong("palantir_cassandra.throttler_table_size_gb", 10L) * 1024L * 1024L * 1024L;
     private static final Logger logger = LoggerFactory.getLogger(CompactionThroughputThrottler.class);
 
+    public static final CompactionThroughputThrottler instance = new CompactionThroughputThrottler();
+    private final Set<UUID> throttledCfs = new HashSet<>();
+
+    private CompactionThroughputThrottler() {}
+
+    public int throttledTableCount()
+    {
+        return throttledCfs.size();
+    }
+
+    public void maybeRemoveThrottledCompaction(CFMetaData cfMetaData)
```

Comment:
Understanding that this is for metric reporting, not really correctness... but curious if adding/removing from this set from different code paths is best? A potential alternative would be to remove from the set when next attempting to get a rate limiter for a cf. I guess the difference would be that with this approach, we'd consider a cf compaction "throttled" between getting the rate limiter and finishing the compaction task, as opposed to considering it as throttled between attempts to get the rate limiter (e.g. until we've actually gone through the disk usage checks again). 

Definitely not as familiar with this code path/workflow, so would love to get your thoughts here.

Reply:
This was my original design but I went with removing this on compaction completion for the purpose of keeping the metric correct and up-to-date. I don't have a strong opinion on how we do this. The trade-off is roughly: relying on the removal during the check for whether throttling is necessary would mean a lot less redundant calls to remove tables from the set, that were never there to begin with **vs** having a very up-to-date metric that does have a lot of redundant calls to Set#remove.